stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -12,28 +12,267 @@
+/// and serialization, please refer to the specific system's documentation.
+///
+/// > Note: For example, you may refer to the [Swift Distributed Actors cluster library](https://github.com/apple/swift-distributed-actors/) documentation,
+/// > which is one example of such feature complete distributed actor system implementation:
+///
+/// ## Implementing a DistributedActorSystem
+///
+/// The following section is dedicated to distributed actor system library authors, and generally can be skipped over
+/// by library users, as it explains the interactions of synthesized code and specific distributed actor system methods
+/// and how they must be implemented.
+///
+/// Methods discussed in this section are generally not intended to be called directly, but instead will have calls
+/// generated to them from distributed actor declarations in appropriate places (such as initializers, `distributed func` calls etc).
+///
+/// ### Assigning and Resigning Actor Identifiers
+///
+/// During a local distributed actor's initialization (i.e. any `init` of a `distributed actor`), the actor system will
+/// be invoked in order to assign an ``ActorID`` for this actor.
+///
+/// A call to ``assignID(_:)`` is made during the initialization of the distributed actor.
+/// The snippet below showcases this, though no guarantees are made at this point about the exact placement of this call.
+///
+/// ```swift
+/// distributed actor ShowcaseIDInit {
+///   // let actorSystem: ActorSystem // synthesized;
+///
+///   // typealias ID = ActorSystem.ActorID
+///   // let id: ID // synthesized; implements `Identifiable.id` requirement
+///
+///   init(actorSystem: ActorSystem) {
+///     self.actorSystem = actorSystem
+///     // ...
+///     // self.id = actorSystem.assignID(Self.self) // synthesized;
+///     // ...
+///   }
+/// }
+/// ```
+///
+/// The result of ``assignID(_:)`` is then directly stored in the synthesized `id` property of the actor.
+///
+/// The actor system should assign _globally unique_ identifiers to types, such that they may be properly resolved
+/// from any process in the distributed actor system. The exact shape of the ``ActorID`` is left up to the library to decide.
+/// It can be as small as an integer based identifier, or as large as a series of key-value pairs identifying the actor.
+///
+/// The actor system must retain a mapping from the ``ActorID`` to the specific actor _instance_ which it is given in
+/// ``actorReady(_)`` in order to implement the ``resolve(id:using:)`` method, which is how incoming and outgoing remote calls are made possible.
+///
+/// Users have no control over this assignment, nor are they allowed to set the `id` property explicitly.
+/// The ``DistributedActor/id`` is used to implement the distributed actor's ``Hashable``, ``Equatable``,
+/// and even ``Codable`` conformance (which is synthesized if and only if the ``ActorID`` is ``Codable`` itself).
```

Comment:
Double \` or single around `Hashable`, `Equatable`, and `Codable`?

Reply:
Double I think, those links should work 🤔

Reply:
Yes, single ` is for code voice and double `` is for links.